### PR TITLE
Dependencies: Remove stencil and reintroduce badge as Angular component

### LINKED ISSUE #3262: [Housekeeping] Remove stencil dependency and reintroduce badge as Angular component
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the housekeeping request
In #1402 we introduced stencil to be able to convert our Angualr components to web-components. The experiment never really caught on, and now seems like a good time to remove stencil as a dependency as we are looking at other alternatives.

## Are there any additional context?
An additional benefit apart from a simpler code base is that we also avoid version mismatches for `jest` between Angular 16 (depends on v29) and Stencil (depends on v27)

### PR DESCRIPTION
**Pull request by RasmusKjeldgaard**
## Which issue does this PR close?

This PR closes #3262

## What is the new behavior?
Badge is now an Angular component again and our core package only exports helpers and shared scss files. 

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

KirbeBadgeModule has been removed and BadgeComponent is now standalone and can be imported directly.

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [x] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] ~~Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).~~

### Review  
- [x] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [ ] ~~Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)~~

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: libs/core/tsconfig.json [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-    "jsx": "react",
-    "jsxFactory": "h",
-    "types": ["jest", "node"]
+    "outDir": "./dist",
+    "rootDir": "src",
+    "skipLibCheck": true,
+    "declaration": true,
+    "declarationDir": "./dist/types"
   },
+  "extends": "../../tsconfig.base.json",
```
Comment (jakobe) [inline libs/core/tsconfig.json]: Please move to top of file to align with rest of code base (as well as it's nice to start with what the file extends and the rest is override to the base settings)

File: libs/designsystem/badge/src/badge.component.scss [hunk 2]
```diff
@@ -20,7 +20,8 @@
     box-shadow: var(--kirby-badge-elevation, none);
   }
 
-  ::slotted(kirby-icon) {
+  /* stylelint-disable selector-pseudo-element-no-unknown */
+  ::ng-deep kirby-icon {
```
Comment (jakobe) [inline libs/designsystem/badge/src/badge.component.scss]: Please use helper method instead to align with rest of code base—and for readability and intent:
```suggestion
  @include utils.slotted(kirby-icon) {
```

**Review by jakobe — CHANGES_REQUESTED** [review]
Nice! A few small things, but almost there 👌🏻 👏🏻

File: libs/designsystem/badge/src/badge.component.scss (hunk 2 above)
Comment (RasmusKjeldgaard) [reply libs/designsystem/badge/src/badge.component.scss]: Nice, fixed in 26154290a 👍

**Review by jakobe — APPROVED** [review]
Great work! 👌🏻

**Pull request merged**